dtls/server.go

```diff
@@ -252,14 +292,20 @@ func (s *Server) Serve(l Listener) error {
 // Stop stops server without wait of ends Serve function.
 func (s *Server) Stop() {
 	s.cancel()
+	s.listenMutex.Lock()
+	l := s.listen
```

Comment:
This seems a bit weird, why bother with the temp variable `l` when you keep holding the mutex anyway?

Perhaps it was means to be:
```
s.listenMutex.Lock()
l := s.listen
s.listen = nil
s.listenMutex.Unlock()
if l != nil {
  l.Close()
}
```

Which avoid calling Close while holding the mutex. But if that's not a problem then `l` can be removed.